### PR TITLE
[NIFI-13056][NIFI-13011] canvas icons, theming, and contrast ratio improvements

### PR DESCRIPTION
**Pull request by scottyaslan**
This continues improving the theme specifically as it pertains to the contrast colors for each of the following icons, images, and or text displayed in the canvas svg:

- the PG version control icons (up to date, locally modified, stale, etc.) on top of the material themed PG header surface (an appropriate text-shadow was added based on light/dark mode)
- the run status icons on the top of the material themed PG detail banner surface
- the not-transmitting and disabled icons when placed on top of the material themed header surfaces of a port or an RPG
- the PG version control icons in the bottom PG breadcrumbs (an appropriate text-shadow was added based on light/dark mode)
- the context menu icons in dark mode
- the listing table header and selected styles
- the navigation header, nifi logo, and the draggable icons
- also cleans up a few typos (`mat-warn` -> `warn-color`) and added missing declarative colors
- comments tooltip indicator on a canvas component fixed for both light and dark modes

### CODE REVIEW

**Comment by mcgilman** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss [hunk 1]
```diff
@@ -26,20 +26,18 @@
...
-
-    // Use the same logic as _navigation.component-theme.scss to determine the bg color for hover
-    $hover-bg: mat.get-color-from-palette($material-theme-primary-palette, 'navbar');
-    @if ($hover-bg) {
-        // Nothing to do here, we have special color from the palette.
-    } @else {
-        // There was not a special value set for the navbar, so we use Angular Material behavior.
-        $hover-bg: mat.get-color-from-palette($material-theme-primary-palette, 'default');
-    }
+    $is-dark: map-get($material-theme-color-config, is-dark);
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss:34]: Not relating to this line directly, in dark mode I do not think there is enough differentiation in the enabled and disabled state of the draggable icons.

**Review by mcgilman — COMMENTED** [review]
Thanks for the PR @scottyaslan! Noticed a couple things on my first pass here that I wanted to pass along.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss (hunk 1 above)
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss:34]: Thanks for the review feedback @mcgilman! This is a particularly difficult thing to get right. The issue is that we need to make sure we have enough contrast with the navigation header background color and the draggable icons. The fact we have a lighter primary color set as the color of the navigation header background color basically forces us to use the opposite color. For example, in light mode we use the 300 hue from the material palette to color the navigation header background so we need to use the 700 hue from that same palette to get enough contrast ration between the two:

Screenshot 2024-04-22 at 9 54 18 AM

Then we also have the hover state to consider: 

Screenshot 2024-04-22 at 9 54 28 AM

As well as the disabled state which up to this point we have opted to just use the default angular material disabled button styles which is a contrast color with no hover state and no pointer mouse cursor:

Screenshot 2024-04-22 at 9 55 19 AM

And of course we need to consider the dark mode versions for any palette:

Screenshot 2024-04-22 at 9 54 05 AM
Screenshot 2024-04-22 at 9 54 40 AM
Screenshot 2024-04-22 at 9 55 06 AM

And then there is also the purple palette that I have been using as a sort of validation that all of these color choices are making sense to be reused in a general sense (meaning that these colors work well with a wide variety of color palettes provided by the angular material team which means creating new themes for nifi becomes much easier as well as maintaining the current themes).

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.ts [hunk 2]
```diff
@@ -155,9 +155,9 @@ export class ControllerServiceTable {
             return 'invalid fa fa-warning';
         } else {
             if (entity.status.runStatus === 'DISABLED') {
-                return 'disabled primary-color icon icon-enable-false';
+                return 'disabled on-surface-default icon icon-enable-false';
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.ts:158]: Do you have a screen shot?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/property-table/editors/nf-editor/_nf-editor.component-theme.scss [hunk 3]
```diff
@@ -31,17 +31,21 @@
...
     $on-surface: utils.get-on-surface($nifi-theme-color-config);
+    $on-surface-lighter: utils.get-on-surface($nifi-theme-color-config, lighter);
 
     .property-editor {
         @include mat.button-density(-1);
 
         .nf-editor {
             .CodeMirror {
-                background-color: $surface;
+                background-color: $nifi-theme-surface-palette-lighter;
```
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/property-table/editors/nf-editor/_nf-editor.component-theme.scss:50]: We absolutely can! I have pushed some updates here in my latest commit.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.ts:158]: Fixed

[the remark this answers is not in the document]
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/controller-service/controller-service-table/controller-service-table.component.ts:158]: Screenshot 2024-04-22 at 12 26 07 PM
Screenshot 2024-04-22 at 12 26 17 PM
Screenshot 2024-04-22 at 12 26 34 PM
Screenshot 2024-04-22 at 12 27 13 PM
Screenshot 2024-04-22 at 12 27 36 PM

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss (hunk 1 above)
Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/ui/canvas/header/new-canvas-item/_new-canvas-item.component-theme.scss:34]: I have added the 'not-allowed' cursor for the disabled state.

**Comment by rfellows** [comment]
reviewing

**Review by rfellows — CHANGES_REQUESTED** [review]
There are some discrepancies between light and dark mode in terms of bulletin/warn styling. In light mode the bulletin icon for the canvas and for processors (and most likely other components) looks good. They use the `warn-color-darker` background with a white color for the icon. However, in dark mode it seems this is not respected and `red` is used as the background and the icon color is white for the canvas but black for the processor.
Screenshot 2024-04-23 at 8 24 05 AM
Screenshot 2024-04-23 at 8 23 52 AM

Also of note, the cluster nodes icon in the top left is no longer indicating a problem when a node is not available. This used to display in the darker warning color, now it doesn't seem to have a color at all (black). This is how it looks in original NiFi:
Screenshot 2024-04-23 at 8 30 47 AM
vs how it looks now:
Screenshot 2024-04-23 at 8 31 47 AM

**Comment by scottyaslan** [comment]
> There are some discrepancies between light and dark mode in terms of bulletin/warn styling. In light mode the bulletin icon for the canvas and for processors (and most likely other components) looks good. They use the `warn-color-darker` background with a white color for the icon. However, in dark mode it seems this is not respected and `red` is used as the background and the icon color is white for the canvas but black for the processor. Screenshot 2024-04-23 at 8 24 05 AM Screenshot 2024-04-23 at 8 23 52 AM
> 

Good catch! The red background colors are correct AFAIK. I will update the bulletin icon color.

> Also of note, the cluster nodes icon in the top left is no longer indicating a problem when a node is not available. This used to display in the darker warning color, now it doesn't seem to have a color at all (black). This is how it looks in original NiFi: Screenshot 2024-04-23 at 8 30 47 AM vs how it looks now: Screenshot 2024-04-23 at 8 31 47 AM

Another great catch! I have updated the warning color of the clustered icon in the flow status menu to be the nifi (theme) warn (palette) darker color.

**Review by rfellows — APPROVED** [review]
LGTM 👍 . Thanks @scottyaslan

**Review by james-elliott — APPROVED** [review]

**Review by rfellows — CHANGES_REQUESTED** [review]
Looks like you left this in "purple" mode

**Review by rfellows — APPROVED** [review]
Looks good, thanks @scottyaslan 👍

**Pull request merged**